qa/tasks/cephadm.py

```diff
@@ -88,24 +88,42 @@ def download_cephadm(ctx, config, ref):
         ref = config.get('cephadm_branch', ref)
         git_url = teuth_config.get_ceph_git_url()
         log.info('Downloading cephadm (repo %s ref %s)...' % (git_url, ref))
-        ctx.cluster.run(
-            args=[
-                'git', 'archive',
-                '--remote=' + git_url,
-                ref,
-                'src/cephadm/cephadm',
-                run.Raw('|'),
-                'tar', '-xO', 'src/cephadm/cephadm',
-                run.Raw('>'),
-                ctx.cephadm,
-                run.Raw('&&'),
-                'test', '-s',
-                ctx.cephadm,
-                run.Raw('&&'),
-                'chmod', '+x',
-                ctx.cephadm,
-            ],
-        )
+        if git_url.startswith('https://github.com/'):
+            # git archive doesn't like https:// URLs, which we use with github.
+            rest = git_url.split('https://github.com/', 1)[1]
```

Comment:
```suggestion
            rest = git_url.split('https://github.com/', 1)[1].split('.git')[0]
```

We need to loose any '`.git` prefix it its there.

Reply:
added a patch to wrok around this.  in general, you should just leave the .git off when scheduling the suite (e.g., ``teuthology-suite --suite-repo https://github.com/liewegas/ceph ...``)